integration/kubernetes/helm-chart/alluxio/templates/csi/nodeplugin.yaml

```diff
@@ -0,0 +1,138 @@
+# The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+# (the "License"). You may not use this work except in compliance with the License, which is
+# available at www.apache.org/licenses/LICENSE-2.0
+#
+# This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+# either express or implied, as more fully set forth in the License.
+#
+# See the NOTICE file distributed with this work for information regarding copyright ownership.
+#
+
+{{ if .Values.csi.enabled -}}
+{{- $name := include "alluxio.name" . }}
+{{- $fullName := include "alluxio.fullname" . }}
+{{- $chart := include "alluxio.chart" . }}
+
+kind: DaemonSet
+apiVersion: apps/v1
+metadata:
+  name: {{ $fullName }}-csi-nodeplugin
+  labels:
+    name: {{ $fullName }}-csi-nodeplugin
+    app: {{ $name }}
+    chart: {{ $chart }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+    role: alluxio-csi-nodeplugin
+spec:
+  selector:
+    matchLabels:
+      app: {{ $name }}
+      role: alluxio-csi-nodeplugin
+      name: {{ $fullName }}-csi-nodeplugin
+  template:
+    metadata:
+      labels:
+        name: {{ $fullName }}-csi-nodeplugin
+        app: {{ $name }}
+        chart: {{ $chart }}
+        release: {{ .Release.Name }}
+        heritage: {{ .Release.Service }}
+        role: alluxio-csi-nodeplugin
+    spec:
+      hostNetwork: true
+      containers:
+        - name: node-driver-registrar
+          image: quay.io/k8scsi/csi-node-driver-registrar:v1.0.2
+          lifecycle:
+            preStop:
+              exec:
+                command: ["/bin/sh", "-c", "rm -rf /registration/alluxio /registration/alluxio-reg.sock"]
+          args:
+            - --v=5
+            - --csi-address=/plugin/csi.sock
+            - --kubelet-registration-path=/var/lib/kubelet/plugins/csi-alluxio-plugin/csi.sock
+          {{- if .Values.csi.driverRegistrar.resources }}
+          resources:
+            {{- if .Values.csi.driverRegistrar.resources.limits }}
+            limits:
+              cpu: {{ .Values.csi.driverRegistrar.resources.limits.cpu }}
+              memory: {{ .Values.csi.driverRegistrar.resources.limits.memory }}
+            {{- end }}
+            {{- if .Values.csi.driverRegistrar.resources.requests }}
+            requests:
+              cpu: {{ .Values.csi.driverRegistrar.resources.requests.cpu }}
+              memory: {{ .Values.csi.driverRegistrar.resources.requests.memory }}
+            {{- end }}
+          {{- end }}
+          env:
+            - name: KUBE_NODE_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: spec.nodeName
+          volumeMounts:
+            - name: plugin-dir
+              mountPath: /plugin
+            - name: registration-dir
+              mountPath: /registration
+        - name: csi-nodeserver
+          securityContext:
+            privileged: true
+            runAsUser: 0
+            runAsGroup: 0
```

Comment:
Is it possible to run as a non-root user?

Reply:
The server need to call `mount` command which require root privilege

Reply:
nit: A general recommendation is for the hard coded variables like this (where the user may want to touch in their deployments), if there's a reason to hard code (not offering an option), we should leave a comment on why this is hard-coded this way.